src/main/host/syscall/handler/timerfd.rs

```diff
@@ -0,0 +1,244 @@
+    syscall_types::{SyscallError, SyscallResult},
+};
+
+impl SyscallHandler {
+    #[log_syscall(/* rv */ libc::c_int, /* clockid */ libc::c_int, /* flags */ libc::c_int)]
+    pub fn timerfd_create(
+        ctx: &mut SyscallContext,
+        clockid: libc::c_int,
+        flags: libc::c_int,
+    ) -> SyscallResult {
+        // Continue only if we support the clockid.
+        check_clockid(clockid)?;
+
+        let Some(flags) = TimerFlags::from_bits(flags) else {
+            log::warn!("Invalid timerfd_create flags: {}", flags);
+            return Err(Errno::EINVAL.into());
+        };
+
+        let mut file_flags = FileStatus::empty();
+        let mut desc_flags = DescriptorFlags::empty();
+
+        if flags.contains(TimerFlags::TFD_NONBLOCK) {
+            file_flags.insert(FileStatus::NONBLOCK);
+        }
+
+        if flags.contains(TimerFlags::TFD_CLOEXEC) {
+            desc_flags.insert(DescriptorFlags::CLOEXEC);
+        }
+
+        let file = TimerFd::new(file_flags);
+        let mut desc = Descriptor::new(CompatFile::New(OpenFile::new(File::TimerFd(file))));
+        desc.set_flags(desc_flags);
+
+        let fd = ctx
+            .objs
+            .process
+            .descriptor_table_borrow_mut()
+            .register_descriptor(desc)
+            .or(Err(Errno::ENFILE))?;
+
+        log::trace!("timerfd_create() returning fd {}", fd);
+
+        Ok(fd.val().into())
+    }
+
+    #[log_syscall(/* rv */ libc::c_int, /* fd */ libc::c_int, /*curr_value*/ *const libc::c_void)]
+    pub fn timerfd_gettime(
+        ctx: &mut SyscallContext,
+        fd: libc::c_int,
+        curr_value_ptr: ForeignPtr<libc::itimerspec>,
+    ) -> SyscallResult {
+        // Get the TimerFd object.
+        let file = get_cloned_file(ctx, fd)?;
+        let File::TimerFd(ref timerfd) = file else {
+            return Err(Errno::EINVAL.into());
+        };
+
+        Self::timerfd_gettime_helper(ctx, timerfd, curr_value_ptr)?;
+
+        Ok(0.into())
+    }
+
+    fn timerfd_gettime_helper(
+        ctx: &mut SyscallContext,
+        timerfd: &Arc<AtomicRefCell<TimerFd>>,
+        value_ptr: ForeignPtr<libc::itimerspec>,
+    ) -> Result<(), Errno> {
+        // Lookup the timer state.
+        let (remaining, interval) = {
+            let borrowed = timerfd.borrow();
+
+            // We return a zero duration if the timer is disarmed.
+            let remaining = match borrowed.get_timer_remaining() {
+                Some(simtime) => simtime,
+                None => SimulationTime::ZERO,
+            };
+            // We return a zero duration if the timer is non-periodic (set to expire only once).
+            let interval = borrowed.get_timer_interval();
+
+            (remaining, interval)
+        };
+
+        // Write the result to the plugin.
+        {
+            // Need a location to write the result.
+            if value_ptr.is_null() {
+                return Err(Errno::EFAULT);
+            }
```

Comment:
We shouldn't need to check for `NULL` here, it should be handled by the eventual memory manager `write()`.

Reply:
(Here is a good example of a place where the mem mgr public api docs could use a few more details.)